ecpprog/ecpprog.c

```diff
@@ -1227,20 +1239,31 @@ int main(int argc, char **argv)
 
 			if (!erase_mode)
 			{
+				uint8_t buffer_flash[256], buffer_file[256], buffer_tx[256];
 				for (int rc, addr = 0; true; addr += rc) {
-					uint8_t buffer[256];
-
 					/* Show progress */
 					fprintf(stderr, "\r\033[0Kprogramming..  %04u/%04lu", addr, file_size);
 
 					int page_size = 256 - (rw_offset + addr) % 256;
-					rc = fread(buffer, 1, page_size, f);
+					rc = fread(buffer_file, 1, page_size, f);
 					if (rc <= 0)
 						break;
+					memcpy(buffer_tx, buffer_file, rc);
+
 					flash_write_enable();
-					flash_prog(rw_offset + addr, buffer, rc);
+					flash_prog(rw_offset + addr, buffer_tx, rc);
 					flash_wait();
 
+					if (!disable_verify && interleaved_verify) {
+						flash_start_read(rw_offset + addr);
+						flash_continue_read(buffer_flash, rc);
+						flash_wait();
```

Comment:
@gregdavill From what I understand from SPI flash this wait here shouldn't be necessary. But without it the second page fails to read correctly.